### PR TITLE
[EPO-527] Install hide code lab extension

### PR DESCRIPTION
**Pull request by cbanek**

### CODE REVIEW

**Comment by cbanek** [comment]
This currently doesn't work, but it seems like it's some kind of version compability problem.  Do not merge until we figure this out.

Output from running:
> /opt/node-v8.10.0-linux-x64/bin/npm pack jupyterlab_hidecode
jupyterlab_hidecode-0.1.0.tgz

Errored, use --debug for full output:
ValueError: This extension does not yet support the current version of JupyterLab.


Conflicting Dependencies:
JupyterLab              Extension        Package
>=0.16.0 <0.17.0        >=0.15.0 <0.16.0 @jupyterlab/application

**Comment by cbanek** [comment]
Also just in case you want to try some things, if you run ./build.sh from the root of the vela repo this should build the image locally without doing a push so you can test out your changes.

**Comment by abbeyyacoe** [comment]
Hey Christine.

Adam took a hacksaw to the bake.sh file. He says there were a lot of redundancies in there you don't need anymore because they're built into the base image now. This works locally for me, but does it break anything else is the question. 

`#!/bin/bash
set -e

# This script runs inside the container to add all our special sauce.

# Install astropixie API library
pip3 install /opt/astropixie
pip3 install /opt/astropixie-widgets

# Install healpy and astroquery
#pip3 install healpy astroquery==0.3.7

# Install ipyaladin
#git clone https://github.com/cds-astro/ipyaladin.git /opt/ipyaladin
#(cd /opt/ipyaladin/js && npm install --unsafe-perm)
#(cd /opt/ipyaladin && pip install -e .)
#(cd /opt/ipyaladin/js && jupyter labextension install)

# Clone our educational notebooks
git clone https://github.com/lsst-epo/investigations.git /opt/investigations

# Gulp in bokeh requires a ridiculously recent NodeJS
#curl -o /opt/node.xz https://nodejs.org/dist/v8.10.0/node-v8.10.0-linux-x64.tar.xz
#(cd /opt && tar xvfJ node.xz)
#export PATH=/opt/node-v8.10.0-linux-x64/bin:$PATH

#rm -rf /tmp/phantomjs
#pip3 uninstall -y bokeh

## Try skipping local bokeh
#git clone https://github.com/bokeh/bokeh.git
#cd bokeh/bokehjs
#npm install --no-save --unsafe-perm
#node ./prepare.js
#cd ..
#python3 setup.py develop --build-js

# Install nbserverproxy.  This allows for URLs like /user/x/proxy/d
# to be routed to port d on the container.
#pip3 install git+https://github.com/jupyterhub/nbserverproxy
#jupyter serverextension enable --py nbserverproxy
jupyter labextension install jupyterlab_hidecode
`

**Comment by cbanek** [comment]
We're going to be diverging from the DM image soon, so we'll need all those redundancies (they won't be redundant after we diverge).  If the script with the installing jupyterlab_hidecode works, I'm fine taking it.  But we need all that stuff to actually work, although it was the hidecode line that was the only thing breaking it.  Does running the hide code with the rest of the script work or fail?

**Comment by abbeyyacoe** [comment]
Got it. With the rest of the script it fails. 

`jupyter labextension install jupyterlab_hidecode` works on it's own but not when you add it to the script. I tried all sorts of hacks to get it to work including something like: 

`git clone https://github.com/lsst-epo/jupyterlab_hidecode.git /opt/investigations/jupyterlab_hidecode && cd /opt/investigations/jupyterlab_hidecode && npm install && npm run build && jupyter labextension install`

As long as the `jupyter labextension` bit is in there, we have trouble, but we need it to enable the extension. 

Tl;dr: I don't think it's an issue with the extension but a compatibility issue with the build script and jupyter labextension. I ran out of ideas for things to try. Any ideas?

**Comment by cbanek** [comment]
Okay, for some reason, when I tried it again yesterday it just worked.  I kept working on it a bit, and with Adam's help found a way to run the notebook locally to verify that the plugin is in there.
